src/utils.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Converts snake_case to camelCase
+ * @param {string} input - snake_case string
+ * @param {string} exclude - `input` that contains `exclude` will be returned unmodified
+ * @returns {string} - camelCase string
+ */
+export function snakeToCamelCase(input: string, exclude = '-'): string {
+  // Return the input string unchanged if there are no underscores
+  // or it includes the `exclude` character.
+  if (!input.includes('_') || input.includes(exclude)) {
+    return input;
+  }
+  // remove trailing/leading underscores
+  let output = input.replace(/^_+/, '').replace(/_+$/, '');
+  let currentIndex = output.indexOf('_');
+  while (currentIndex !== -1) {
+    output =
+      output.slice(0, currentIndex) +
+      output[currentIndex + 1].toUpperCase() +
+      output.slice(currentIndex + 2);
+    currentIndex = output.indexOf('_', currentIndex);
+  }
+  return output;
+}
```

Comment:
Could this just be the following?

```
export function snakeToCamelCase(input: string): string {
  return input.replace(/_+(\w?)/g, (_, p, o) => o == 0 ? p : p.toUpperCase());
}
```
It probably could be even simpler is we didn't worry about leading or double underscores, which will not show up in our web service response keys anyway.